### PR TITLE
[ROCm] Minor updates to sync some of the ROCm fork contents 

### PR DESCRIPTION
**Pull request by deven-amd**
see individual commit messages for details.

-------

/cc @whchung @chsigg @cheshire

### CODE REVIEW

File: tensorflow/compiler/tests/matrix_triangular_solve_op_test.py [hunk 1]
```diff
@@ -87,6 +87,13 @@ def testBasicNotActuallyTriangular(self):
       self._VerifyTriangularSolveCombo(a.astype(dtype), b.astype(dtype))
 
   def testBasicComplexDtypes(self):
+
+    if xla_test.test.is_built_with_rocm():
+      # TODO(rocm)
```
Comment (cheshire) [inline tensorflow/compiler/tests/matrix_triangular_solve_op_test.py]: The TF convention is to have usernames of responsible people inside the TODO block, not a feature name. It might be better to remove this comment TODO entirely

**Review by cheshire — DISMISSED** [review]

File: tensorflow/compiler/tests/matrix_triangular_solve_op_test.py (hunk 1 above)
Comment (deven-amd) [reply tensorflow/compiler/tests/matrix_triangular_solve_op_test.py]: @cheshire 

removed the comment, and also fixed the linter errors. 

please re-approve. thanks

File: tensorflow/compiler/xla/debug_options_parsers_test.cc [hunk 2]
```diff
@@ -26,8 +26,8 @@ namespace xla {
 
 // Test that the xla_backend_extra_options flag is parsed correctly.
 TEST(DebugOptionsFlags, ParseXlaBackendExtraOptions) {
-  std::unordered_map<string, string> test_map;
-  string test_string = "aa=bb,cc,dd=,ee=ff=gg";
+  std::unordered_map<std::string, std::string> test_map;
+  std::string test_string = "aa=bb,cc,dd=,ee=ff=gg";
```
Comment (cheshire) [inline tensorflow/compiler/xla/debug_options_parsers_test.cc]: While this change is correct, if required, it should be done in a separate dedicated PR.

File: tensorflow/compiler/xla/refcounting_hash_map.h [hunk 3]
```diff
@@ -22,6 +22,7 @@ limitations under the License.
 #include "absl/container/node_hash_map.h"
 #include "absl/memory/memory.h"
 #include "absl/synchronization/mutex.h"
+#include "tensorflow/core/platform/logging.h"
```
Comment (cheshire) [inline tensorflow/compiler/xla/refcounting_hash_map.h]: Is this required?

File: tensorflow/compiler/xla/refcounting_hash_map_test.cc [hunk 4]
```diff
@@ -77,7 +77,7 @@ TEST(RefcountingHashMapTest, CustomFactory) {
 
 TEST(RefcountingHashMapTest, ForEachEmpty) {
   RefcountingHashMap<int, int> m;
-  int64 count = 0;
+  int64_t count = 0;
```
Comment (cheshire) [inline tensorflow/compiler/xla/refcounting_hash_map_test.cc]: Why this change, `int64` is used everywhere throughout the codebase?

File: tensorflow/compiler/xla/service/dynamic_padder_test.cc [hunk 5]
```diff
@@ -15,6 +15,7 @@ limitations under the License.
 
 #include "tensorflow/compiler/xla/service/dynamic_padder.h"
 
+#include "absl/strings/str_replace.h"
```
Comment (cheshire) [inline tensorflow/compiler/xla/service/dynamic_padder_test.cc]: If required, this needs to be a separate PR.

File: tensorflow/core/nccl/nccl_manager_test.cc [hunk 6]
```diff
@@ -663,8 +663,8 @@ TYPED_TEST(NcclManagerTest, MultiNodeBroadcast) {
   this->RunMultiNodeBroadcastTest(num_nodes, num_ranks_per_node,
                                   /*src_node=*/0, /*src_local_rank=*/0,
                                   /*in_place=*/true);
-#endif
 }
+#endif
```
Comment (cheshire) [inline tensorflow/core/nccl/nccl_manager_test.cc]: How was this even compiling before?

**Review by cheshire — CHANGES_REQUESTED** [review]
Hi,

In general, we don't do changes which do many different little things with titles similar to "minor updates". The important bit here seems to be skipping tests on ROCM, could we stick to that?

Another PR could rename Cuda2DLaunchConfig to Gpu2DLaunchConfig, and maybe CudaLaneId to GpuLaneId (though probably just LaneId is better), and maybe even GPU_1D_KERNEL_LOOP macro (though again probably better to just drop GPU_ altogether).

I'm not sure if other changes are even necessary.

File: tensorflow/core/nccl/nccl_manager_test.cc (hunk 6 above)
Comment (deven-amd) [reply tensorflow/core/nccl/nccl_manager_test.cc]: This test has not been compiling (in ROCm mode) since this commit broke it : 
https://github.com/tensorflow/tensorflow/commit/d65a5f1bdf5dc536efe9cfcdd64acc2d6273b6e2

File: tensorflow/compiler/xla/debug_options_parsers_test.cc (hunk 2 above)
Comment (deven-amd) [reply tensorflow/compiler/xla/debug_options_parsers_test.cc]: this change, and others below are all required to make the corresponding testcases build in ROCm mode, they will all fail compilation without the change.

The changes do not seem ROCm specific, so am not convinced whether these tests are working to begin

Comment (deven-amd) [reply tensorflow/compiler/xla/debug_options_parsers_test.cc]: compile errors without the fix

```
...
external/com_google_googletest/googletest/include/gtest/internal/gtest-port.h:881:23: note:   'testing::internal::string'
 typedef ::std::string string;
                       ^
tensorflow/compiler/xla/debug_options_parsers_test.cc:29:33: error: 'test_map' was not declared in this scope
   unordered_map<string, string> test_map;
                                 ^
tensorflow/compiler/xla/debug_options_parsers_test.cc:30:10: error: expected ';' before 'test_string'
   string test_string = "aa=bb,cc,dd=,ee=ff=gg";
...
  ```

File: tensorflow/compiler/xla/refcounting_hash_map_test.cc (hunk 4 above)
Comment (deven-amd) [reply tensorflow/compiler/xla/refcounting_hash_map_test.cc]: compile error without this change

```
ERROR: /root/tensorflow/tensorflow/compiler/xla/BUILD:928:1: C++ compilation of rule '//tensorflow/compiler/xla:refcounting_hash_map_test' failed (Exit 1)
tensorflow/compiler/xla/refcounting_hash_map_test.cc: In member function 'virtual void xla::{anonymous}::RefcountingHashMapTest_ForEachEmpty_Test::TestBody()':
tensorflow/compiler/xla/refcounting_hash_map_test.cc:80:3: error: 'int64' was not declared in this scope
   int64 count = 0;
...
```

File: tensorflow/compiler/xla/refcounting_hash_map.h (hunk 3 above)
Comment (deven-amd) [reply tensorflow/compiler/xla/refcounting_hash_map.h]: this may not be required...will remove it

File: tensorflow/compiler/xla/service/dynamic_padder_test.cc (hunk 5 above)
Comment (deven-amd) [reply tensorflow/compiler/xla/service/dynamic_padder_test.cc]: compile error without the header

```
INFO: Deleting stale sandbox base /root/.cache/bazel/_bazel_root/efb88f6336d9c4a18216fb94287b8d97/sandbox
ERROR: /root/tensorflow/tensorflow/compiler/xla/service/BUILD:2311:1: C++ compilation of rule '//tensorflow/compiler/xla/service:dynamic_padder_test_gpu' failed (Exit 1)
tensorflow/compiler/xla/service/dynamic_padder_test.cc: In member function 'virtual void xla::{anonymous}::ExecutionTest_ScatterUpdate_Test::TestBody()':
tensorflow/compiler/xla/service/dynamic_padder_test.cc:266:7: error: 'StrReplaceAll' is not a member of 'absl'
       absl::StrReplaceAll(hlo_text, {{"INDICES_BOUND", "2"}});
       ^
tensorflow/compiler/xla/service/dynamic_padder_test.cc:281:7: error: 'StrReplaceAll' is not a member of 'absl'
       absl::StrReplaceAll(hlo_text, {{"INDICES_BOUND", "4"}});
...
```

**Review by deven-amd — COMMENTED** [review]
I can break this up into multiple PRs, once we have resolved all the feedback.

**Comment by cheshire** [comment]
Let's break this up into multiple PRs and discuss one by one. It would be also good to understand why you are seeing a compilation failure but we don't. What toolchain do you use?

File: tensorflow/core/nccl/nccl_manager_test.cc (hunk 6 above)
Comment (deven-amd) [reply tensorflow/core/nccl/nccl_manager_test.cc]: @cheshire, separated out this commit into PR #36017

**Comment by deven-amd** [comment]
@cheshire I have broken up this PR into the following 6

#36017 
#36018 
#36019 
#36030 
#36031 
#36032 

Closing out this PR.
